app/tasks.py

```diff
@@ -1,95 +1,56 @@
-s3_user_image_dir = S3_BUCKET_CONFIG["USER_ASSETS"]
 
 
 @celery.task()
-def process_images(filename, tmp_dir, dir_to_save):
+def image_process(folder_name, s3_dir):
     """
-    This function processes the image files, using the PIL image library. The filename argument is
-    used to create writable binary object, the object is overwritten by the data of the image downloaded
-    from s3 using boto3 client. Then the image processed using PIL is uploaded back to s3 using boto3 client.
-    The name of this object the filename argument. The image is deleted in the temp dir for cleanup.
+    Resize the image files, using the PIL image library
     """
-    try:
-        img_obj = open(filename, "wb")
-        s3.download_fileobj(bucket, f"{s3_temp_dir + tmp_dir}{filename}", img_obj)
-        pil_image_obj = Image.open(filename)
-        pil_image_obj.thumbnail((800, 800))
-        pil_image_obj.save(filename)
-
-        with open(filename, "rb") as data_obj:
-            image_uploader(data_obj, s3_prop_image_dir + dir_to_save + filename)
-        os.remove(filename)
-        delete_img_objs(bucket, s3_temp_dir + tmp_dir, file_obj=filename)
-    except Exception as e:
-        return e
+    img_data = redis_client.hgetall(folder_name)
+    for img_name in img_data.keys():
+        byte_key_to_str = img_name.decode("utf-8")
+        # Get image byte object from redis with r_key as object key
+        byte_img_obj = redis_client.hget(folder_name, byte_key_to_str)
+        decoded = Image.open(io.BytesIO(byte_img_obj))
+        decoded.thumbnail((800, 800))
+        decoded.save(byte_key_to_str, format="JPEG")
+        upload_to_s3.delay(byte_key_to_str, s3_dir, folder_name)
 
 
 @celery.task()
-def update_prop_images(temp_dir, image_file_list, prop_id):
-    from app.base.models import Property
-
-    images_list = update_property_images(temp_dir, image_file_list, prop_id)
-    img_list_to_json = json.dumps(images_list)
-    Property.update_property_images(images_list[0], img_list_to_json, prop_id)
-    return "Images Updated"
-
-
-@celery.task()
-def update_profile_image(dir_name, filename, content_type):
-    s3.upload_fileobj(
-        open(filename, "rb"),
-        bucket,
-        f"{s3_user_image_dir}{dir_name}{filename}",
-        ExtraArgs={
-            "ACL": "public-read",
-            "ContentType": content_type
-        }
-    )
-    os.remove(filename)
-
-
-@celery.task()
-def image_uploader(image_file, path_to_save):
+def upload_to_s3(img_obj, s3_dir, folder_name):
+    """
+    Upload to S3 and delete the image from redis.
+    """
     try:
         s3.upload_fileobj(
-            image_file,
+            open(img_obj, "rb"),
             bucket,
-            path_to_save,
-            ExtraArgs={
-                "ACL": "public-read",
-                "ContentType": "image/jpeg"
-            }
+            f"{s3_dir}{folder_name}/{img_obj}",
+            ExtraArgs={"ACL": "public-read", "ContentType": "image/jpeg"},
```

Comment:
same here format issues `jpeg` limitations

Reply:
I have added an expression to add the `ContentType` of an image according to its extension. If the file ends with `.jpg` or `.jpeg`, the `ContentType` should be `image/jpeg` otherwise `image/png`. While testing I figured that not specifying the `ContentType` can lead to Mime Type issues where the browser does not recognize the image or the browser downloads the image when clicked instead of a view.